src/OpenDiffix.Core/Parser.fs

```diff
@@ -51,19 +51,18 @@ module QueryParser =
 
   let alias = word "AS" >>. identifier
 
-  // This custom numbers parser is needed as both pint64 and pfloat are eager
-  // to the point of not being possible to combine. pint64 would parse 1.2 as 1,
-  // and pfloat would parse 1 as 1.0.
+  let numberFormat =
+    NumberLiteralOptions.AllowMinusSign
+    ||| NumberLiteralOptions.AllowFraction
+    ||| NumberLiteralOptions.AllowExponent
+
   let number =
-    pint64 .>>. opt (pchar '.' >>. many (pchar '0') .>>. opt pint32) .>> spaces
-    |>> fun (wholeValue, decimalPartOption) ->
-          match decimalPartOption with
-          | None -> Expression.Integer wholeValue
-          | Some (leadingZeros, Some decimalValue) ->
-            let divisor = List.length leadingZeros + 1
-            let decimalPart = (float decimalValue) / (float <| pown 10 divisor)
-            Expression.Float(float wholeValue + decimalPart)
-          | Some (_leadingZeros, None) -> Expression.Float(float wholeValue)
+    numberLiteral numberFormat "number" .>> spaces
+    |>> fun nl ->
+          if nl.IsInteger then
+            Expression.Integer(int64 nl.String)
+          else
+            Expression.Float(float nl.String)
```

Comment:
Does `float nl.String` perform culture sensitive parsing?

Reply:
according to some SO answer - no, but I couldn't find a more reliable source

Reply:
but, I've found `FParsec` to use `System.Double.Parse(nl.String, System.Globalization.CultureInfo.InvariantCulture)` instead, in its own `pfloat`, so I guess that would be more consistent

Reply:
I decided to force push just this little detail

Reply:
I might have already asked this question. I think we override thread culture to force it to the invariant one.

Reply:
^ I checked and that's the case. BUT, we do that in `main`, therefore a custom CLI client such as desktop can miss that particular detail and result in weird behavior.